src/main/java/counter/service/BookingService.java

```diff
@@ -24,9 +27,16 @@
     @Autowired
     BookingRepo bookingRepository;
 
+    @Autowired
+    FlightRepo flightRepository;
+
     public void saveBooking(Booking booking, Jwt principal) {
         System.out.println(booking + " : " + principal.getClaims());
         booking.setBookingAgent(principal.getSubject());
+        Optional<Flight> flightOptional = flightRepository.findById(booking.getFlight().getFlightId());
+        if (!flightOptional.isPresent())
```

Comment:
if you do flightOptional.get() doesn't it already throw an exception you can catch when nothing is file.